qemu/tests/drive_mirror.py

```diff
@@ -62,7 +62,8 @@ def get_target_image(self):
 
     def get_device(self):
         params = super(DriveMirror, self).parser_test_args()
-        image_file = storage.get_image_filename(params, self.data_dir)
+        image_name = "%s.%s" % (params["image_name"], params["image_format"])
+        image_file = utils_misc.get_path(self.data_dir, image_name)
```

Comment:
What's the reason to replace original approach?

Reply:
@luckyh Original approach from storage module will always return the image name generated by guest os params, which is 'images/rhel75-64-virtio-scsi.qcow2' alike. While for drive_mirror.with_installation, I use a new image with different name, so use this new approach to get correct image file path.

Reply:
They should do the same thing if you set the `params` right, otherwise, this issue is caused by the wrong params rather than the "wrong" approach.

Btw, I'm also curious about why it gets params by using `super()` not `self`?